test/disconnect-spec.js

```diff
@@ -32,6 +35,31 @@ describe('disconnections', function () {
         .then(docker.start)
         .catch(done);
     });
+
+    it('should retry producing only as configured', (done) => {
```

Comment:
Please use async-await

Reply:
I made the new test async, I will open another pr for the others.